### ISSUE TITLE
Unable to successfully test connector in GCC

### ISSUE THREAD

**Issue by carrollamanda2** [bguidinger/Xrm.ReportRenderer#35]
I am able to successfully create and test this custom connector in us.flow.  When attempting the same setup in gov.flow, i consistently get a bad authorization token (400) error.

Error from token exchange: Bad authorization token. The access token is from wrong audience or resource.

Is there anything that needs to be done differently for this custom connector to work in GCC??

Thanks!
Amanda

**Comment by bguidinger** [comment]
Hi Amanda,

The solution is hardcoded to go to the commercial authentication endpoint
right now (login.microsoftonline.com).  I believe it just needs to be
updated to go to login.microsoftonline.us.

I don't have time/capacity to make this change right now.  If you're a
developer, or you're working with a developer, they could make this change
here:
https://github.com/bguidinger/Xrm.ReportScheduler/blob/15cc2b84ed336c227359209c450ddb308a9729de/src/Plugins/Reports/Renderer.cs#L117

**Comment by carrollamanda2** [comment]
Thank you!

We are actually using commercial azure and our Dynamics is in GCC.  When I create the Custom Connector in commercial flow, I am able to test the connector successfully on the test tab of the custom connector setup.  When I create the custom connector in gcc flow, the test fails on the test tab of the custom connector.  At this point I think it's just using the parameters entered in the setup and the json file, correct?

**Comment by Yocomd** [comment]
Wondering if you ever resolved this?  GCC uses Commercial Azure, so the change above doesn't resolve it.

**Comment by Yocomd** [comment]
We figured it out.  Need to change the tenant from "common" to the guid for your tenant and need to make sure the return url is added to your app registration.  Actually, turns out we only solved part of it.  We were able to create and test the connector, but when trying to use it in a flow, we get a "Something went wrong" error because SecureConfig is null.  And editing again because I realized I just need to update the username and password in the managed solution config!
